### PR TITLE
RHDEVDOCS-3217 Tweaks to JSON parsing topics

### PR DESCRIPTION
**Pull request by rolfedh**
- Aligned team: Dev Tools
- For branches: 4.7+
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-3217
- Direct link to doc preview: https://deploy-preview-35248--osdocs.netlify.app/openshift-enterprise/latest/logging/cluster-logging-enabling-json-logging.html
- FYI: @alanconway
- SME review: @sichvoge  
- QE review: @kabirbhartiRH 
- Peer review: @ 	tbd
- <All reviews complete. Please merge now>

### CODE REVIEW

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc [hunk 1]
```diff
@@ -1,15 +1,15 @@
 [id="cluster-logging-configuration-of-json-log-data-for-default-elasticsearch_{context}"]
 = Configuring JSON log data for Elasticsearch
 
-When forwarding JSON logs to an Elasticsearch log store, you must create an index for each format if the JSON log entries _have different formats_.
+If your JSON logs have more than one schema, storing them in a single index might cause type conflicts and cardinality problems. To avoid that, you must configure the `ClusterLogForwarder` custom resource (CR) to group each schema into a single output definition. This way, each schema is forwarded to a separate index.
```
Comment (sichvoge) [inline modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc:4]: I would change to:
> If your JSON logs follow more than one schema ...

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+
+If you forward to Red Hat's OpenShift Logging managed Elasticsearch, it will try and generate a new index based on your configuration.
```
Comment (sichvoge) [inline modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc:6]: Should that be captured into a "[NOTE]" tag?

**Comment by sichvoge** [comment]
Can you also adjust the first paragraph in https://deploy-preview-35248--osdocs.netlify.app/openshift-enterprise/latest/logging/cluster-logging-enabling-json-logging.html#cluster-logging-forwarding-json-logs-to-the-default-elasticsearch_cluster-logging-enabling-json-logging.

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
 [IMPORTANT]
 ====
-You must create a separate index for each different JSON log format. Otherwise, forwarding different formats to the same index can cause type conflicts and cardinality problems.
+To avoid performance issues associated with having too many indices, keep the  number of possible schemas low by standardizing to common schemas.
```
Comment (sichvoge) [inline modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc:10]: can we say 
> ..., we recommend to keep ...

Comment (sichvoge) [reply modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc:10]: seems there is also an additional whitespace in between "the" and "number"

**Review by sichvoge — CHANGES_REQUESTED** [review]

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc (hunk 1 above)
Comment (rolfedh) [reply modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc:4]: Updated. Thanks.

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc (hunk 2 above)
Comment (rolfedh) [reply modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc]: Updated. Thanks.

File: modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc (hunk 3 above)
Comment (rolfedh) [reply modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc]: Updated. But, unfortunately, I'm not allowed to use "we" or "recommend". If we want to recommend something, we simply tell them to do it, like we've done here.  I'll soften it by saying "consider keeping..."

Comment (rolfedh) [reply modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc]: Updated. Thanks.

**Comment by rolfedh** [comment]
> Can you also adjust the first paragraph in https://deploy-preview-35248--osdocs.netlify.app/openshift-enterprise/latest/logging/cluster-logging-enabling-json-logging.html#cluster-logging-forwarding-json-logs-to-the-default-elasticsearch_cluster-logging-enabling-json-logging.

Updated to say "follow schemas"

**Comment by rolfedh** [comment]
@sichvoge Thank you for your comments! This helps. Please review, comment, and/or approve.

**Review by sichvoge — APPROVED** [review]

File: modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc [hunk 4]
```diff
@@ -1,7 +1,14 @@
 [id="cluster-logging-forwarding-json-logs-to-the-default-elasticsearch_{context}"]
 = Forwarding JSON logs to the Elasticsearch log store
 
-For the Elasticsearch log store that OpenShift Logging manages, you must create a different index for each format in advance if your JSON log entries _have different formats_. Otherwise, forwarding different formats to the same index can cause type conflicts and cardinality problems.
+For an Elasticsearch log store, if your JSON log entries _follow different schemas_, configure the `ClusterLogForwarder` custom resource (CR) to group each JSON schema into a single output definition. This way, Elasticsearch uses a separate index for each schema.
```
Comment (bergerhoffer) [inline modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc:4]: I don't think the emphasis on "follow different schemas" is necessary. But I see it was similarly done in the previous text, so I won't hold up the PR over it or anything.

File: modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc [hunk 5]
```diff
@@ -1,7 +1,14 @@
...
+
+[IMPORTANT]
+====
+Because forwarding different schemas to the same index can cause type conflicts and cardinality problems, you must do perform this configuration before you forward data to the Elasticsearch store.
```
Comment (bergerhoffer) [inline modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc:8]: Typo "do perform"

File: modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc (hunk 4 above)
Comment (rolfedh) [reply modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc:4]: We don't want users to fuss over this procedure if their JSON log entries all follow one schema. I'd welcome suggestions on a better way to word this.

File: modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc (hunk 5 above)
Comment (rolfedh) [reply modules/cluster-logging-forwarding-json-logs-to-the-default-elasticsearch.adoc:8]: Updated. Thanks.

**Pull request merged**